recipes/_configure_indexes.rb

```diff
@@ -39,9 +39,10 @@
   daily_mb = hash.delete('_maxDailyDataSizeMB')
   padding = hash.delete('_dataSizePaddingPercent')
   default_config = config.fetch('default', {})
+  s2_enabled_index = hash.delete('_is_s2Index') || default_config['_is_s2Index']
   # _noGenerateTstatsHomePath is false  by default.
   no_gentstat = hash.delete('_noGenerateTstatsHomePath') || default_config['_noGenerateTstatsHomePath']
-  if %i[index default].include?(stanza_type) && daily_mb && !hash.key?('maxTotalDataSizeMB')
+  if %i[index default].include?(stanza_type) && daily_mb && !hash.key?('maxTotalDataSizeMB') && !hash.key?('maxGlobalDataSizeMB')
```

Comment:
This allows for a potentially unexpected interaction.  If they set the maxTotalDataSize and enable s2, but don't set the maxGlobalDataSize then the global data size won't be auto generated.  It's not something that should ever happen, but if it does I think we could handle it better.  Possibly something like:
`%i[index default].include?(stanza_type) && daily_mb && ((!s2_enabled_index && !hash.key?('maxTotalDataSizeMB')) || (s2_enabled_index && !hash.key?('maxGlobalDataSizeMB')))` though there may be a better way.